packages/opencensus-instrumentation-grpc/src/grpc.ts

```diff
@@ -0,0 +1,478 @@
+}
+const grpcClient = requireModule('grpc', 'src/client.js');
+
+
+//
+// Types definitions
+//
+export type GrpcModule = typeof grpcModule;
+type MakeClientConstructor = typeof grpcModule.makeGenericClientConstructor;
+type RegisterMethod = typeof grpcModule.Server.prototype.register;
+
+type Status = {
+  code: number, details: string; metadata: grpcModule.Metadata;
+};
+
+type ServerCall = typeof grpcModule.ServerUnaryCall|
+                  typeof grpcModule.ServerReadableStream|
+                  typeof grpcModule.ServerWriteableStream|
+                  typeof grpcModule.ServerDuplexStream;
+
+type ServerCallWithMeta = ServerCall&{
+  metadata: grpcModule.Metadata;
+  status: Status
+}
+&EventEmitter;
+
+export type SendUnaryDataCallback =
+    (error: grpcModule.ServiceError,
+     // tslint:disable-next-line:no-any
+     value?: any, trailer?: grpcModule.Metadata,
+     flags?: grpcModule.writeFlags) => void;
+
+
+type HandlerSet = {
+  // tslint:disable-next-line:no-any
+  func: grpcModule.handleCall<any, any>;
+  // tslint:disable-next-line:no-any
+  serialize: grpcModule.serialize<any>;
+  // tslint:disable-next-line:no-any
+  deserialize: grpcModule.deserialize<any>;
+  type: string;
+};
+
+
+/** gRPC instrumentation plugin for Opencensus */
+export class GrpcPlugin extends classes.BasePlugin {
+  /**
+   * Span grpc attributes
+   */
+  static readonly ATTRIBUTE_GRPC_KIND = 'grpc.kind';  // SERVER or CLIENT
+  static readonly ATTRIBUTE_GRPC_METHOD = 'grpc.method';
+  static readonly ATTRIBUTE_GRPC_STATUS_CODE = 'grpc.status_code';
+  static readonly ATTRIBUTE_GRPC_ERROR_NAME = 'grpc.error_name';
+  static readonly ATTRIBUTE_GRPC_ERROR_MESSAGE = 'grpc.error_message';
+
+  private logger: types.Logger;
+
+  /** Constructs a new GrpcPlugin instance. */
+  constructor() {
+    super('grpc');
+  }
+
+  /**
+   * Patches gRPC incoming and outcoming request functions.
+   * @param moduleExports The gRPC module exports
+   * @param tracer A tracer instance to create spans on.
+   * @param version The package version.
+   */
+  applyPatch(moduleExports: GrpcModule, tracer: types.Tracer, version: string) {
+    this.setPluginContext(moduleExports, tracer, version);
+    this.logger = tracer.logger || logger.logger('debug');
+
+    this.logger.debug('applying pacth to %s@%s', this.moduleName, this.version);
+
+    shimmer.wrap(
+        moduleExports.Server.prototype, 'register', this.patchServer());
+    shimmer.wrap(grpcClient, 'makeClientConstructor', this.patchClient());
+
+    return moduleExports;
+  }
+
+
+  /** Unpatches all gRPC patched function. */
+  applyUnpatch(): void {
+    shimmer.unwrap(this.moduleExports.Server.prototype, 'register');
+    shimmer.unwrap(grpcClient, 'makeClientConstructor');
+  }
+
+
+  /**
+   * Returns a function that patches the gRPC server register function in order
+   * to create trace spans for gRPC service methods.
+   * @returns {Function}  The register patched function.
```

Comment:
`{Function}` annotation not needed here.

Reply:
Done!